zetaclient/chains/solana/signer/signer.go

```diff
@@ -214,21 +218,31 @@ func (signer *Signer) TryProcessOutbound(
 }
 
 // signTx creates and signs solana tx containing provided instruction with relayer key.
-func (signer *Signer) signTx(ctx context.Context, inst *solana.GenericInstruction) (*solana.Transaction, error) {
+func (signer *Signer) signTx(ctx context.Context, inst *solana.GenericInstruction, limit uint64) (*solana.Transaction, error) {
 	// get a recent blockhash
 	recent, err := signer.client.GetLatestBlockhash(ctx, rpc.CommitmentFinalized)
 	if err != nil {
 		return nil, errors.Wrap(err, "getLatestBlockhash error")
 	}
 
+	// if limit is provided, prepend compute unit limit instruction
+	var instructions []solana.Instruction
+	if limit > SolanaMaxComputeBudget {
+		limit = SolanaMaxComputeBudget
+	}
+
+	if limit > 0 && limit <= SolanaMaxComputeBudget {
```

Comment:
Not sure why having `&& limit <= SolanaMaxComputeBudget` ?

Reply:
Do we need this check , if limit is already set to a maximum of `SolanaMaxComputeBudget`